make/hotspot/lib/JvmOverrideFiles.gmk

```diff
@@ -86,6 +86,11 @@ else ifeq ($(call isTargetOs, macosx), true)
     # for the clang bug was still needed.
     BUILD_LIBJVM_loopTransform.cpp_CXXFLAGS := $(CXX_O_FLAG_NONE)
 
+    # See JDK-8340341
+    ifeq "$(firstword $(subst ., ,$(CXX_VERSION_NUMBER)))" "16"
+      BUILD_LIBJVM_stackMapTable.cpp_CXXFLAGS := "-O1"
```

Comment:
As Julian points out, we don't use quotes in makefiles (as make doesn't treat or interpret quotes a quotes, they are just literal characters).
```suggestion
    ifeq ($(firstword $(subst ., ,$(CXX_VERSION_NUMBER))), 16)
      BUILD_LIBJVM_stackMapTable.cpp_CXXFLAGS := -O1
```

I also kind of agree with David Holmes, this compiler has shown that we can't trust it. At the very least, we should include a big warning somewhere, probably in configure.